From 20ee2c902ae03accbdd1ec772f4fa1248e6ac2c3 Mon Sep 17 00:00:00 2001
From: Jo-Philipp Wich <jo@mein.io>
Date: Wed, 18 Jan 2017 03:40:50 +0100
Subject: [PATCH] perl-www-curl: fix build against curl >= 7.50

Import a proposed upstream bug fix to allow building against recent curl
versions. Fixes the following error observed by the buildbots:

    curlopt-constants.c:129:49: error: 'CURL_STRICTER' undeclared (first use in this function)
                 if (strEQ(name, "STRICTER")) return CURL_STRICTER;

Upstream bug: https://rt.cpan.org/Public/Bug/Display.html?id=117793

Signed-off-by: Jo-Philipp Wich <jo@mein.io>
---
 .../patches/101-skip-preprocessor-symbol.path | 34 +++++++++++++++++++
 1 file changed, 34 insertions(+)
 create mode 100644 lang/perl-www-curl/patches/101-skip-preprocessor-symbol.path

diff --git a/lang/perl-www-curl/patches/101-skip-preprocessor-symbol.path b/lang/perl-www-curl/patches/101-skip-preprocessor-symbol.path
new file mode 100644
index 000000000..34221c0d2
--- /dev/null
+++ b/lang/perl-www-curl/patches/101-skip-preprocessor-symbol.path
@@ -0,0 +1,34 @@
+From 0be0223422e6e5f4091c6e4e058d213623eed105 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
+Date: Mon, 12 Sep 2016 14:40:44 +0200
+Subject: [PATCH] Skip preprocessor symbol only CURL_STRICTER
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+CURL_STRICTER leaked into curl-constants.c when building against
+curl-7.50.2. This is a preprocessor only macro without a value.
+
+CPAN RT#117793
+
+Signed-off-by: Petr Písař <ppisar@redhat.com>
+---
+ Makefile.PL | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/Makefile.PL b/Makefile.PL
+index f9170bb..ad2bd3d 100644
+--- a/Makefile.PL
++++ b/Makefile.PL
+@@ -127,7 +127,7 @@ if (!defined($curl_h)) {
+     close H;
+ 
+     for my $e (sort @syms) {
+-       if($e =~ /(OBSOLETE|^CURL_EXTERN|_LAST\z|_LASTENTRY\z)/) {
++       if($e =~ /(OBSOLETE|^CURL_EXTERN|^CURL_STRICTER\z|_LAST\z|_LASTENTRY\z)/) {
+           next;
+        }
+        my ($group) = $e =~ m/^([^_]+_)/;
+-- 
+2.7.4
+
-- 
2.25.1

